File: search-domain/src/main/java/com/example/search/domain/analytics/LatencyPercentiles.java
package com.example.search.domain.analytics;

/**
 * 분석 구간의 검색 응답 latency 분포 — ms 단위.
 *
 * <p>{@code sampleSize} 는 통계 신뢰도 — 1k 미만이면 p99 가 흔들리므로 운영자가 표시 여부를 판단.</p>
 */
public record LatencyPercentiles(
        long p50,
        long p95,
        long p99,
        long sampleSize
) {

    public LatencyPercentiles {
        if (p50 < 0 || p95 < 0 || p99 < 0) {
            throw new IllegalArgumentException("percentile 음수 불가");
        }
        if (sampleSize < 0) {
            throw new IllegalArgumentException("sampleSize 음수 불가");
        }
    }

    public static LatencyPercentiles empty() {
        return new LatencyPercentiles(0, 0, 0, 0);
    }
}

File: search-domain/src/main/kotlin/com/example/search/domain/analytics/LatencyPercentiles.kt
package com.example.search.domain.analytics

/**
 * 분석 구간의 검색 응답 latency 분포 — ms 단위.
 *
 * [sampleSize] 는 통계 신뢰도 — 1k 미만이면 p99 가 흔들리므로 운영자가 표시 여부를 판단.
 */
@JvmRecord
data class LatencyPercentiles(
    val p50: Long,
    val p95: Long,
    val p99: Long,
    val sampleSize: Long
) {
    init {
        require(p50 >= 0 && p95 >= 0 && p99 >= 0) { "percentile 음수 불가" }
        require(sampleSize >= 0) { "sampleSize 음수 불가" }
    }

    companion object {
        @JvmStatic
        fun empty(): LatencyPercentiles = LatencyPercentiles(0, 0, 0, 0)
    }
}
